File: server-scala-play-server/app/api/FavoriteApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.AccountResponse
import model.BigDecimal
import model.SearchResponse
import model.SirqulResponse
import model.WrappedResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@Singleton
class FavoriteApiController @Inject()(cc: ControllerComponents, api: FavoriteApi) extends AbstractController(cc) {
  /**
    * POST /api/:version/favorite/create?deviceId=[value]&accountId=[value]&favoritableId=[value]&favoritableType=[value]&latitude=[value]&longitude=[value]
    */
  def addFavorite(version: BigDecimal): Action[AnyContent] = Action { request =>
    def executeApi(): WrappedResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val favoritableId = request.getQueryString("favoritableId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("favoritableId", "query string")
        }
        
      val favoritableType = request.getQueryString("favoritableType")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("favoritableType", "query string")
        }
        
      val latitude = request.getQueryString("latitude")
        .map(value => value.toDouble)
        
      val longitude = request.getQueryString("longitude")
        .map(value => value.toDouble)
        
      api.addFavorite(version, favoritableId, favoritableType, deviceId, accountId, latitude, longitude)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/:version/favorite/delete?deviceId=[value]&accountId=[value]&favoriteId=[value]&favoritableId=[value]&favoritableType=[value]
    */
  def deleteFavorite(version: BigDecimal): Action[AnyContent] = Action { request =>
    def executeApi(): SirqulResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val favoriteId = request.getQueryString("favoriteId")
        .map(value => value.toLong)
        
      val favoritableId = request.getQueryString("favoritableId")
        .map(value => value.toLong)
        
      val favoritableType = request.getQueryString("favoritableType")
        
      api.deleteFavorite(version, deviceId, accountId, favoriteId, favoritableId, favoritableType)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/:version/favorite/get?deviceId=[value]&accountId=[value]&favoriteId=[value]&latitude=[value]&longitude=[value]
    */
  def getFavorite(version: BigDecimal): Action[AnyContent] = Action { request =>
    def executeApi(): WrappedResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val favoriteId = request.getQueryString("favoriteId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("favoriteId", "query string")
        }
        
      val latitude = request.getQueryString("latitude")
        .map(value => value.toDouble)
        
      val longitude = request.getQueryString("longitude")
        .map(value => value.toDouble)
        
      api.getFavorite(version, favoriteId, deviceId, accountId, latitude, longitude)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/:version/favorite/search?deviceId=[value]&accountId=[value]&connectionAccountId=[value]&favoritableType=[value]&secondaryType=[value]&keyword=[value]&sortField=[value]&descending=[value]&start=[value]&limit=[value]&activeOnly=[value]&returnFullResponse=[value]&latitude=[value]&longitude=[value]
    */
  def searchFavorites(version: BigDecimal): Action[AnyContent] = Action { request =>
    def executeApi(): SearchResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val connectionAccountId = request.getQueryString("connectionAccountId")
        .map(value => value.toLong)
        
      val favoritableType = request.getQueryString("favoritableType")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("favoritableType", "query string")
        }
        
      val secondaryType = request.getQueryString("secondaryType")
        
      val keyword = request.getQueryString("keyword")
        
      val sortField = request.getQueryString("sortField")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("sortField", "query string")
        }
        
      val descending = request.getQueryString("descending")
        .map(value => value.toBoolean)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("descending", "query string")
        }
        
      val start = request.getQueryString("start")
        .map(value => value.toInt)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("start", "query string")
        }
        
      val limit = request.getQueryString("limit")
        .map(value => value.toInt)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("limit", "query string")
        }
        
      val activeOnly = request.getQueryString("activeOnly")
        .map(value => value.toBoolean)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("activeOnly", "query string")
        }
        
      val returnFullResponse = request.getQueryString("returnFullResponse")
        .map(value => value.toBoolean)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("returnFullResponse", "query string")
        }
        
      val latitude = request.getQueryString("latitude")
        .map(value => value.toDouble)
        
      val longitude = request.getQueryString("longitude")
        .map(value => value.toDouble)
        
      api.searchFavorites(version, favoritableType, sortField, descending, start, limit, activeOnly, returnFullResponse, deviceId, accountId, connectionAccountId, secondaryType, keyword, latitude, longitude)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/:version/favorite/whois?deviceId=[value]&accountId=[value]&latitude=[value]&longitude=[value]&favoritableId=[value]&favoritableType=[value]&keyword=[value]&start=[value]&limit=[value]
    */
  def whoHasFavorited(version: BigDecimal): Action[AnyContent] = Action { request =>
    def executeApi(): List[AccountResponse] = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val latitude = request.getQueryString("latitude")
        .map(value => value.toDouble)
        
      val longitude = request.getQueryString("longitude")
        .map(value => value.toDouble)
        
      val favoritableId = request.getQueryString("favoritableId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("favoritableId", "query string")
        }
        
      val favoritableType = request.getQueryString("favoritableType")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("favoritableType", "query string")
        }
        
      val keyword = request.getQueryString("keyword")
        
      val start = request.getQueryString("start")
        .map(value => value.toInt)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("start", "query string")
        }
        
      val limit = request.getQueryString("limit")
        .map(value => value.toInt)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("limit", "query string")
        }
        
      api.whoHasFavorited(version, favoritableId, favoritableType, start, limit, deviceId, accountId, latitude, longitude, keyword)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}

File: server-java-play-framework/app/controllers/FavoriteApiController.java
package controllers;

import apimodels.AccountResponse;
import java.math.BigDecimal;
import apimodels.SearchResponse;
import apimodels.SirqulResponse;
import apimodels.WrappedResponse;

import com.typesafe.config.Config;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import play.libs.Files.TemporaryFile;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import javax.validation.Valid;
import com.typesafe.config.Config;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-09T20:49:20.380387445Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
public class FavoriteApiController extends Controller {
    private final FavoriteApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Config configuration;

    @Inject
    private FavoriteApiController(Config configuration, FavoriteApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }

    @ApiAction
    public Result addFavorite(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valuefavoritableId = request.getQueryString("favoritableId");
        Long favoritableId;
        if (valuefavoritableId != null) {
            favoritableId = Long.parseLong(valuefavoritableId);
        } else {
            throw new IllegalArgumentException("'favoritableId' parameter is required");
        }
        String valuefavoritableType = request.getQueryString("favoritableType");
        String favoritableType;
        if (valuefavoritableType != null) {
            favoritableType = valuefavoritableType;
        } else {
            throw new IllegalArgumentException("'favoritableType' parameter is required");
        }
        String valuelatitude = request.getQueryString("latitude");
        Double latitude;
        if (valuelatitude != null) {
            latitude = Double.parseDouble(valuelatitude);
        } else {
            latitude = null;
        }
        String valuelongitude = request.getQueryString("longitude");
        Double longitude;
        if (valuelongitude != null) {
            longitude = Double.parseDouble(valuelongitude);
        } else {
            longitude = null;
        }
        return imp.addFavoriteHttp(request, version, favoritableId, favoritableType, deviceId, accountId, latitude, longitude);
    }

    @ApiAction
    public Result deleteFavorite(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valuefavoriteId = request.getQueryString("favoriteId");
        Long favoriteId;
        if (valuefavoriteId != null) {
            favoriteId = Long.parseLong(valuefavoriteId);
        } else {
            favoriteId = null;
        }
        String valuefavoritableId = request.getQueryString("favoritableId");
        Long favoritableId;
        if (valuefavoritableId != null) {
            favoritableId = Long.parseLong(valuefavoritableId);
        } else {
            favoritableId = null;
        }
        String valuefavoritableType = request.getQueryString("favoritableType");
        String favoritableType;
        if (valuefavoritableType != null) {
            favoritableType = valuefavoritableType;
        } else {
            favoritableType = null;
        }
        return imp.deleteFavoriteHttp(request, version, deviceId, accountId, favoriteId, favoritableId, favoritableType);
    }

    @ApiAction
    public Result getFavorite(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valuefavoriteId = request.getQueryString("favoriteId");
        Long favoriteId;
        if (valuefavoriteId != null) {
            favoriteId = Long.parseLong(valuefavoriteId);
        } else {
            throw new IllegalArgumentException("'favoriteId' parameter is required");
        }
        String valuelatitude = request.getQueryString("latitude");
        Double latitude;
        if (valuelatitude != null) {
            latitude = Double.parseDouble(valuelatitude);
        } else {
            latitude = null;
        }
        String valuelongitude = request.getQueryString("longitude");
        Double longitude;
        if (valuelongitude != null) {
            longitude = Double.parseDouble(valuelongitude);
        } else {
            longitude = null;
        }
        return imp.getFavoriteHttp(request, version, favoriteId, deviceId, accountId, latitude, longitude);
    }

    @ApiAction
    public Result searchFavorites(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valueconnectionAccountId = request.getQueryString("connectionAccountId");
        Long connectionAccountId;
        if (valueconnectionAccountId != null) {
            connectionAccountId = Long.parseLong(valueconnectionAccountId);
        } else {
            connectionAccountId = null;
        }
        String valuefavoritableType = request.getQueryString("favoritableType");
        String favoritableType;
        if (valuefavoritableType != null) {
            favoritableType = valuefavoritableType;
        } else {
            throw new IllegalArgumentException("'favoritableType' parameter is required");
        }
        String valuesecondaryType = request.getQueryString("secondaryType");
        String secondaryType;
        if (valuesecondaryType != null) {
            secondaryType = valuesecondaryType;
        } else {
            secondaryType = null;
        }
        String valuekeyword = request.getQueryString("keyword");
        String keyword;
        if (valuekeyword != null) {
            keyword = valuekeyword;
        } else {
            keyword = null;
        }
        String valuesortField = request.getQueryString("sortField");
        String sortField;
        if (valuesortField != null) {
            sortField = valuesortField;
        } else {
            throw new IllegalArgumentException("'sortField' parameter is required");
        }
        String valuedescending = request.getQueryString("descending");
        Boolean descending;
        if (valuedescending != null) {
            descending = Boolean.valueOf(valuedescending);
        } else {
            throw new IllegalArgumentException("'descending' parameter is required");
        }
        String valuestart = request.getQueryString("start");
        Integer start;
        if (valuestart != null) {
            start = Integer.parseInt(valuestart);
        } else {
            throw new IllegalArgumentException("'start' parameter is required");
        }
        String valuelimit = request.getQueryString("limit");
        Integer limit;
        if (valuelimit != null) {
            limit = Integer.parseInt(valuelimit);
        } else {
            throw new IllegalArgumentException("'limit' parameter is required");
        }
        String valueactiveOnly = request.getQueryString("activeOnly");
        Boolean activeOnly;
        if (valueactiveOnly != null) {
            activeOnly = Boolean.valueOf(valueactiveOnly);
        } else {
            throw new IllegalArgumentException("'activeOnly' parameter is required");
        }
        String valuereturnFullResponse = request.getQueryString("returnFullResponse");
        Boolean returnFullResponse;
        if (valuereturnFullResponse != null) {
            returnFullResponse = Boolean.valueOf(valuereturnFullResponse);
        } else {
            throw new IllegalArgumentException("'returnFullResponse' parameter is required");
        }
        String valuelatitude = request.getQueryString("latitude");
        Double latitude;
        if (valuelatitude != null) {
            latitude = Double.parseDouble(valuelatitude);
        } else {
            latitude = null;
        }
        String valuelongitude = request.getQueryString("longitude");
        Double longitude;
        if (valuelongitude != null) {
            longitude = Double.parseDouble(valuelongitude);
        } else {
            longitude = null;
        }
        return imp.searchFavoritesHttp(request, version, favoritableType, sortField, descending, start, limit, activeOnly, returnFullResponse, deviceId, accountId, connectionAccountId, secondaryType, keyword, latitude, longitude);
    }

    @ApiAction
    public Result whoHasFavorited(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valuelatitude = request.getQueryString("latitude");
        Double latitude;
        if (valuelatitude != null) {
            latitude = Double.parseDouble(valuelatitude);
        } else {
            latitude = null;
        }
        String valuelongitude = request.getQueryString("longitude");
        Double longitude;
        if (valuelongitude != null) {
            longitude = Double.parseDouble(valuelongitude);
        } else {
            longitude = null;
        }
        String valuefavoritableId = request.getQueryString("favoritableId");
        Long favoritableId;
        if (valuefavoritableId != null) {
            favoritableId = Long.parseLong(valuefavoritableId);
        } else {
            throw new IllegalArgumentException("'favoritableId' parameter is required");
        }
        String valuefavoritableType = request.getQueryString("favoritableType");
        String favoritableType;
        if (valuefavoritableType != null) {
            favoritableType = valuefavoritableType;
        } else {
            throw new IllegalArgumentException("'favoritableType' parameter is required");
        }
        String valuekeyword = request.getQueryString("keyword");
        String keyword;
        if (valuekeyword != null) {
            keyword = valuekeyword;
        } else {
            keyword = null;
        }
        String valuestart = request.getQueryString("start");
        Integer start;
        if (valuestart != null) {
            start = Integer.parseInt(valuestart);
        } else {
            throw new IllegalArgumentException("'start' parameter is required");
        }
        String valuelimit = request.getQueryString("limit");
        Integer limit;
        if (valuelimit != null) {
            limit = Integer.parseInt(valuelimit);
        } else {
            throw new IllegalArgumentException("'limit' parameter is required");
        }
        return imp.whoHasFavoritedHttp(request, version, favoritableId, favoritableType, start, limit, deviceId, accountId, latitude, longitude, keyword);
    }

}
